lib/connection.js

```diff
@@ -905,45 +917,62 @@ function _setClient(conn, client, options, dbName) {
  * Closes the connection
  *
  * @param {Boolean} [force] optional
+ * @param {Boolean} [closeAndClean] optional
```

Comment:
I would personally prefer to make the first parameter an options-object, with attributes force and closeAndClean. And if first parameter is of type boolean, the use it as a value for force. Currently it is backwards breaking and thus imho not acceptable.
Maybe also rename it. We call the method close so closeAndClean is doubled. So you can remove "closeAnd". On the other hand: what is meant with "clean"? Cleaning is here meant to be freeing of used resources. What is the resource? Sockets? Memory? Maybe should be called freeSockets or whatever it does.

Reply:
Ah wait, should be backwards compatible. But still the naming is strange.

Reply:
@Uzlopak 

`force` is a type of string or an object and is sent to native-mongodb-driver, whereas `closeAndClean` has been introduced for cleaning up closed MongoDB connection. `cleanAndClose`  introduced with the close method should indicate that the user wants to clean that closed connection from memory and won't be used for re-connection.

Reply:
I just want to express my concern that "cleanAndClose" is a bad name. It should be destroyConnections or something like that to indicate the actual functionality.

Reply:
Thank you @vkarpov15 @Uzlopak for your reviews on this PR.

I would be implementing the following changes. 

1. Instead of having a parameter in the `close` method, the user would call `conn.destroy()` 
2. conn.destroy()`  would take similar parameters to that of `close`, but in this method, the `destroyConnAfterClose` flag would be set to true.
3. Same would be checked in `openUri` method.

So from the user point of view, user would be calling `conn.destroy()` instead of `conn.close()`

@vkarpov15 @Uzlopak Please verify my understanding is correct, and I will be pushing in the changes by EOD today. 
Thanks